app/views/users/show.html.erb

```diff
@@ -1 +1,14 @@
-<h1>User Profile #<%= @id %></h1>
\ No newline at end of file
+<section>
+    <%= render "shared/user_card", user: @user %>
+
+    <div class="user-bio">
+        <h3>Bio</h3>
+        <p><%= @user.bio %></p>
+    </div>
+
+    <% @user.recent_posts.each do |post| %>
+        <%= render "shared/post_card", post: post %>
+    <% end %>
+
+    <%= link_to "See All Posts", user_posts_path(@user.id), class: "link-btn" %>
```

Comment:
Actually, this can't be done because `user_posts_path` is responsible for displaying all posts for a given user and it doesn't require a second parameter for the post.

You can see that in this route:
```sh
user_posts GET    /users/:user_id/posts(.:format)
```

Reply:
Yeah sure but If you want to display a single post for a user, you can use the user_post_path helper method and pass in the user and post as parameters like this: `<%= link_to post.title, user_post_path(@user, post) %>
` , I was not  able to get the user post

Reply:
Inside `views/shared/_post_card.html.erb` file, I added the following code to link to a specific post for a given user.
```
    <h3><%= link_to post.title, user_post_path(user_id: post.author.id, id: post.id), class: "post-card-title" %></h3>
```
I think this is what you meant.